src/Idle.DataAccess/Fields/FrameworkBase.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Idle.DataAccess.Buffs;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Idle.DataAccess.Fields
+{
+    /// <summary>
+    /// Abstract class for all frameworks
+    /// </summary>
+    public abstract class FrameworkBase : ModelBase, IDescriptive, IBuffing, IProgress, ICost, IRequirement
+    {
+        // Descriptive Properties
+        /// <summary>
+        /// Name of the framework
+        /// </summary>
+        public abstract string Name { get; }
+        /// <summary>
+        /// Description of the framework
+        /// </summary>
+        public abstract string Description { get; }
+        /// <summary>
+        /// Difficulty of the framework
+        /// </summary>
+        public abstract ModelBase.Difficulty Difficulty { get; }
+
+
+
+        // Buff Properties
+        private float _xpMult = 1f;
+        /// <summary>
+        /// Base XP Multiplier of this framework, the starter value may vary depending on the framework
+        /// </summary>
+        public float XPMult { get => _xpMult; set => _xpMult = value; }
+
+        private float _speedMult = 1f;
+        /// <summary>
+        /// Base Speed Multiplier of this framework, the starter value may vary depending on the framework
+        /// </summary>
+        public float SpeedMult { get => _speedMult; set => _speedMult = value; }
+        /// <summary>
+        /// What buffs the framework can be affected by
+        /// </summary>
+        public abstract List<BuffBase> Buffs { get; }
```

Comment:
when we will use buffers in the feature then i would not define them as abstract.
Because when I understood correctly, then buffs can be only added at runtime and they expire, is this correct?
If yes then they will never be defined and initiated in the concretion.

But I would remove this IBuffing for now at all, as it causes unimplemented stuff.
We could make a clean prototype and add features like buffing for example.
At that point we will also see how extensible and flexible the designed and written app is.
Does this make sense?

Reply:
For the buff part, I thought we were already going to provide what buffs a language can be affected by during concretion.
However, you are right, we should get rid of it for now.